net/FlatBuffers/ByteBuffer.cs

```diff
@@ -106,7 +116,9 @@ protected void WriteLittleEndian(int offset, int count, ulong data)
 
         protected ulong ReadLittleEndian(int offset, int count)
         {
+#if !BYTEBUFFER_NO_BOUNDS_CHECK
```

Comment:
I guess sticking the #if inside of AssertOffsetAndLength is not going to yield the performance gain? (C# compiler doesn't cull empty methods?)

Also please indent the #if if you don't mind, so much more readable.